tests/validators/blocks/test_block_validator.py

```diff
@@ -93,3 +93,22 @@ def test_invalid_self_reference():
         }
     ]
     assert validator.errors == expected_errors
+
+
+def test_invalid_list_collector_repeating_blocks_placeholder_references_same_block():
```

Comment:
I think we need to add a test of `SectionValidator` since that's where you added your new validation.

Reply:
Moved the test from test_block_validator to test_section_validator and calling validation from an instance of SectionValidator 👍